art/attacks/evasion/projected_gradient_descent/projected_gradient_descent_tensorflow_v2.py

```diff
@@ -226,17 +234,26 @@ def _generate_batch(
 
         :param x: An array with the original inputs.
         :param targets: Target values (class labels) one-hot-encoded of shape `(nb_samples, nb_classes)`.
+        :param mask: An array with a mask to be applied to the adversarial perturbations. Shape needs to be
+                     broadcastable to the shape of x. Any features for which the mask is zero will not be adversarially
+                     perturbed.
+        :param eps: Maximum perturbation that the attacker can introduce.
+        :param eps_step: Attack step size (input variation) at each iteration.
```

Comment:
"param mask" is now twice in the docstring, let's keep the one in line 237 which is consistent with other attacks

Reply:
Hi- the redundant docstring has been removed now!